### PR TITLE
Rework Auth state management - Match react <> react native

### LINKED ISSUE #207: React - Standardize auth in RN + React. Drop context in favor of Zustand
# Checklist
- [x] Remove context
- [x] Add zustand and source of auth
- [x] properly manage auth in login/signup with the new store
- [x] Add the token in the axios interceptor (rest)
- [x] Do similar step in gql variant

### PR DESCRIPTION
**Pull request by lakardion**
## What this does

Closes #207 

## How to test

Add user steps to achieve desired functionality for this feature.

### CODE REVIEW

File: {{cookiecutter.project_slug}}/clients/web/react/src/pages/log-in.tsx [hunk 1]
```diff
@@ -10,26 +10,28 @@ import { Link, Navigate, useLocation, useNavigate } from 'react-router-dom'
...
   const [error, setError] = useState(autoError ? true : false)
-  const { updateToken, updateUserId } = useAuth()
+  const { changeToken, changeUserId } = useAuth.use.actions()
   const { createFormFieldChangeHandler, form } = useTnForm<TLoginForm>()
   const navigate = useNavigate()
 
 {% if cookiecutter.use_graphql == 'y' -%}
   const [logIn] = useMutation(LOG_IN, {
     onCompleted: (data: { tokenAuth: { token: string } }) => {
-      localStorage.setItem('auth-token', data.tokenAuth.token)
```
Comment (lakardion) [inline {{cookiecutter.project_slug}}/clients/web/react/src/pages/log-in.tsx:30]: zustand automatically syncs with localstorage thanks to the `persist` middleware

File: {{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/settings.py [hunk 2]
```diff
@@ -412,11 +412,8 @@
 
 CORS_ALLOWED_ORIGINS = ["https://{{ cookiecutter.project_slug|replace('_', '-') }}-staging.herokuapp.com", "https://{{ cookiecutter.project_slug|replace('_', '-') }}.herokuapp.com"]
 {% if cookiecutter.client_app.lower() != 'none' -%}
-CORS_ALLOWED_ORIGINS.append("http://localhost:8089")
+CORS_ALLOWED_ORIGINS.append("http://localhost:8080")
 {% endif -%}
-{% if cookiecutter.use_graphql == 'y' -%}
-CORS_ALLOWED_ORIGINS.append("http://localhost:3000")
-{%- endif -%}
```
Comment (oudeismetis) [inline {{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/settings.py:419]: I like this change ha

File: {{cookiecutter.project_slug}}/clients/web/react/src/app.tsx [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 export const AppRoot = () => {
   return (
     <Router>
+      {% if cookiecutter.use_graphql=='y' -%}
       <AuthProvider>
+      {% endif -%}
         <AppRoutes />
+      {% if cookiecutter.use_graphql=='y' -%}
       </AuthProvider>
+      {% endif -%}
```
Comment (oudeismetis) [inline {{cookiecutter.project_slug}}/clients/web/react/src/app.tsx:16]: petty but, space between please. `use_graphql == 'y'`

**Comment by oudeismetis** [comment]
Publish is failing
`SyntaxError: App.tsx: /home/runner/work/tn-spa-bootstrapper/tn-spa-bootstrapper/my_project/mobile/App.tsx: Use process(css).then(cb)`

**Review by paribaker — APPROVED** [review]

**Pull request merged**